### PR TITLE
chore: add tests for postgres_users.go

### PR DESCRIPTION
**Pull request by carolinaecalderon**
## Description
Add tests for `postgres_users.go` and `postgres_agentusergroups.go`

## Test Plan
Pass the test included.

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

## Ticket
DET-9840

### CODE REVIEW

File: master/internal/user/postgres_users_intg_test.go
Comment (stoksc) [inline master/internal/user/postgres_users_intg_test.go:21]: I've run into this lately. It feels better to just add a TestMain IMO, rather than repeat this all over.

File: master/internal/user/postgres_users_intg_test.go
Comment (stoksc) [inline master/internal/user/postgres_users_intg_test.go:27]: this is much more compact than the tests I wrote for templates. I weighed this vs what I did and went with what I did because the test failures much more clearly identify what feature in template failed and the tests are easier to maintain when refactoring since a change to one API affects one small test, not a lot to do with one large test.

wdyt?

File: master/internal/user/postgres_users_intg_test.go
Comment (stoksc) [inline master/internal/user/postgres_users_intg_test.go:7]: https://google.github.io/styleguide/go/decisions#import-renaming

File: master/internal/user/postgres_users_intg_test.go
Comment (carolinaecalderon) [reply master/internal/user/postgres_users_intg_test.go]: see new commit!

File: master/internal/user/postgres_users_intg_test.go
Comment (stoksc) [inline master/internal/user/postgres_users_intg_test.go:21]: I would just define this in the scope it is needed, otherwise refactoring stuff to actually use it / test with a ctx will be really awkward.

**Review by stoksc — COMMENTED** [review]
just a few questions

File: master/internal/user/postgres_users_intg_test.go
Comment (carolinaecalderon) [reply master/internal/user/postgres_users_intg_test.go:255]: yeah see https://github.com/determined-ai/determined/pull/7886, waiting on a resolution to that first

File: master/internal/user/postgres_users_intg_test.go
Comment (carolinaecalderon) [reply master/internal/user/postgres_users_intg_test.go:77]: ByID returns `FullUser` type, it's more straightforward to do direct query and just get `User` type

File: master/internal/user/postgres_users_intg_test.go
Comment (NicholasBlaskey) [inline master/internal/user/postgres_users_intg_test.go:138]: maybe could set ```tt.updatedUser.DisplayName = "dontupdate"```, and assert that it isn't updated because it isn't in the field mask 

displayName is just an example for a field

File: master/internal/user/postgres_users_intg_test.go
Comment (NicholasBlaskey) [inline master/internal/user/postgres_users_intg_test.go:175]: could use ```require.ErrorAs``` here for whatever bun uses for the not found error

The only concern is that our tests are passing but they are somehow broken and we didn't know

**Review by NicholasBlaskey — COMMENTED** [review]
LGTM this looks great, thanks for writing tests

I'm only not hitting approve since I will leave it to @stoksc

**Review by stoksc — APPROVED** [review]
lgtm modulo final comments

File: master/internal/user/postgres_users_intg_test.go
Comment (carolinaecalderon) [reply master/internal/user/postgres_users_intg_test.go:308]: unfortunately no, I can include a standardization in https://github.com/determined-ai/determined/pull/7886 if you'd like -- should it be int32 or int?

Comment (stoksc) [reply master/internal/user/postgres_users_intg_test.go:308]: this was just a comment, not on you to fix.

**Pull request merged**
